src/librustc/arena.rs

```diff
@@ -36,6 +36,7 @@ macro_rules! arena_types {
                 rustc::mir::BodyAndCache<$tcx>
             >,
             [] tables: rustc::ty::TypeckTables<$tcx>,
+            [] borrowck_result: rustc::mir::BorrowCheckResult<$tcx>,
```

Comment:
Can you not add `decode` here instead of the `load_cached` on the query?

Reply:
Maybe, I was just copying `TypeckTables`.